libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -197,7 +197,7 @@ void GCS_MAVLINK::send_battery_status(const AP_BattMonitor &battery, const uint8
                                     battery.get_cell_voltages(instance).cells, // cell voltages
                                     battery.has_current(instance) ? battery.current_amps(instance) * 100 : -1, // current
                                     battery.has_current(instance) ? battery.current_total_mah(instance) : -1, // total current
-                                    -1, // joules used
+                                    battery.has_current(instance) ? battery.consumed_wh(instance) * 36 : -1, // consumed energy in hJ (hecto-Joules)
```

Comment:
While checking `battery.has_current()` seems like a valid approach this does not consider the case of the smart batteries which aren't implementing the Wh counting. You could either implement it in the smart battery drivers (which is probably more work then you want to do I'm guessing), or the approach of either checking that consumed_wh(instanace) is not 0 would also work. The other option you have is to just set the default value of the consumed_Wh in the state to -1 which would result in the analog batteries working easily, and provide the correct MAVLink message for the smart batteries. (I would recommend the latter option as it's the easiest/most robust)